docs/use_cases/recommender_systems.md

```diff
@@ -1,15 +1,111 @@
+
+Beside MF, the same approach can be used with [EASE](https://arxiv.org/abs/1905.03375), for example, or with an explicitly graph-based approach for recommendations such as [PageRank](https://scikit-network.readthedocs.io/en/latest/use_cases/recommendation.html). In fact, with PageRank, the walks would include the dummy nodes.
+
+One question remains: which weights (ratings) to put for the dummy user interactions? We suggest you first start with low weights, and see how the quality of the recommendations changes.
+
+**Minimal Code**
+
+Here is some minimal Python code, demonstrating the addition of dummy users, one for each category:
+
+``` python
+import implicit
+import numpy as np
+import scipy.sparse as sp
+
+# Creating a synthetic user-item interaction matrix
+user_item = sp.coo_matrix((np.random.randint(0, 2, 100), (np.random.randint(0, 10, 100), np.random.randint(0, 10, 100))))
+
+# Creating synthetic item categories
+item_categories = ['A', 'B', 'A', 'C', 'B', 'C', 'A', 'B', 'C', 'A']
+
+# Create dummy users for each unique category
+unique_categories = list(set(item_categories))
+dummy_users = {category: i + user_item.shape[0] for i, category in enumerate(unique_categories)}
+
+# Extend the user-item matrix with dummy interactions
+extended_user_item = sp.vstack([
+    user_item,
+    sp.coo_matrix((len(unique_categories), user_item.shape[1]))
+]).tocsr()  # Ensure the matrix is in CSR format
+
+# Add dummy interactions
+# For each item, add an interaction with its corresponding dummy user
+rows, cols, data = [], [], []
+for item_idx, category in enumerate(item_categories):
+    rows.append(dummy_users[category])
+    cols.append(item_idx)
+    data.append(1)  # or another value indicating the strength of the dummy interaction
+
+# Add dummy interactions to the extended_user_item matrix
+dummy_interactions = sp.coo_matrix((data, (rows, cols)), shape=extended_user_item.shape)
+
+print(rows, cols, data)
+
+extended_user_item += dummy_interactions
+
+print(extended_user_item)
+
+# Fit the ALS model
+model = implicit.als.AlternatingLeastSquares(factors=50)
+model.fit(extended_user_item.T)  # Transpose to item-user format as implicit library expects item-user matrix
+
+# Generate recommendations
+# Note: Ensure not to recommend dummy users
+user_id = 0  # Example user ID
+recommended_items = model.recommend(user_id, extended_user_item[user_id], N=5, filter_already_liked_items=True)
+
+recommended_items
+```
+
+### Potential problems
+
+Obviously, the addition of dummy nodes increases the computational and memory complexity, but in most cases this is a negligible compromise, given the scarcity of the side information. In an audio-on-demand platform, with 250K items in the catalog, there were a few hundred item categories only.
```

Comment:
from previous comments, not yet addressed

"In an audio-on-demand platform":
Give context. Suddenly, we are presented with a specific case, and in the past tense, whereas you are discussing generally, and in the present tense. It makes the discussion hard to understand.

instead of "In an audio-on-demand platform, with 250K items in the catalog, there were a few hundred item categories only."
say "For example, when we performed ALS MF on an audio-on-demand platform with a catalog of 250k items, there were only a few hundred item categories."

Also, is this LightFM? If so, say that when you introduce it and throughout. Instead I see LightFM discussed in the concluding paragraph, as if it's a separate instance.

Your next paragraph (line 93) gives this context, but you don't connect the two. Instead, below, it seems like a separate instance. I would merge the two paragraphs and put them under a some kind of representative section title, like "### Real-world use case."
See above for note about how the heading "### Potential problems" doesn't accurately describe what follows.